### DISCUSSION TITLE
MRTK3 Preview 16 Release

### DISCUSSION THREAD

**Discussion by AMollis** [microsoft/MixedRealityToolkit-Unity#11586] [Releases]
![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/e0814ab9-b058-4a8a-a105-9b50c4e8b401)

The MRTK team is thrilled to announce the pre.16 release for MRTK3! This release includes bug fixes for our solver scripts and enhancements for our release pipelines. We also brought MRTK2’s non-native keyboard to MRTK3, with some minor improvements to the keyboard’s preview text.

As always, we welcome your feedback on these changes and look forward to reaching MRTK3 GA soon.

# Changelog for MRTK3 pre.16

## [pre.16] - 2023-5-30 - [`b4b1684`](https://github.com/microsoft/MixedRealityToolkit-Unity/commit/b4b16846089de0fb5349e31bc26b8609e4bd5b04)

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). The content is mirrored at [the changelog page of the Microsoft docs website](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/changelogs/mrtk3-pre-16-changelog), for now.

### Added

- Added Non-native Keyboard to MRTK3 ([#11549](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11549))
- Added Dictation to Non-Native Keyboard ([#11560](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11560))
- Added Multi-Line, RTL Text, and Better Caret Support to Keyboard Preview and Non-Native Keyboard ([#11566](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11566))
- Added a release candidate pipeline ([#11562](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11562))

### Changed

- Updated MRTK3 package dependencies that weren't updated to use com.unity.xr.interaction.toolkit 2.3.0 ([#11542](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11542))
- Modified the CI pipeline to produce packages with a date code and an optional prerelease tag ([#11546](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11546))
- Update Non-Canvas Dialog Prefab ([#11563](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11563))

### Fixed

- Fixed edge case duplication bug in VirtualizedScrollRectList ([#11538](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11538))
- Fixed SolverHandler issues when handedness flag is set to "everything" (i.e. 0xFFFFFFFF or -1) ([#11543](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11543))
- Fixed MRTK3's Tap to Place solver to work with any hand and interactor ([#11545](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11545))
- Fixed execution order with ProximityLights ([#11575](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11575))

**Comment by LocalJoost** [comment]
Hi, in the sources it says you are now using Unity 2021.3.21f1. However, even an empty app with only a basic scene with nothing in it but the XR MRTK Rig crashes on startup when deployed to a HoloLens. If I use 2021.3.16f1 it all works.

**Reply by AMollis** [reply]
@LocalJoost, the MRTK3 samples are built and tested with 2021.3.21f1.  Unfortunately, we haven't seen the crash you're describing.  Can you share a call stack and/or exception of the failure? Can you share the list of packages and versions you are using?  Also, as usual, sharing a simple project that reproduces the issue will help expediate resolution of the failure.

Please open a new issue if you believe this is an actionable issue: 

https://github.com/microsoft/MixedRealityToolkit-Unity/issues/new/choose

**Reply by LocalJoost** [reply]
Here you go https://github.com/microsoft/MixedRealityToolkit-Unity/issues/11604. Repro'ed just 5 minutes ago

**Reply by AMollis** [reply]
Thank you @LocalJoost ... Issue is related to a stack overflow caused by com.unity.inputsystem v1.5.0.  Updating to v1.5.1 should resolve your issue.

I am going to update MRTK's min input system version, to avoid issues in the future.

- https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11609

**Comment by AMollis** [comment]
We discovered as issue introduced with our release pipeline.  This issue causes MRTK3 dependency downloads to fail if you don't manually select the dependencies from the Feature Tool.  We're working on a fix for pre.17

- https://github.com/microsoft/MixedRealityToolkit-Unity/issues/11607

Sorry for the inconvenience.

**Comment by 719664746** [comment]
您好 我在Unity2021.3.14中使用TextAccessibility的时候并没有描述中的效果出现，是版本有问题吗，还是需要打包出来以后才能看到效果
